apiv2/retention/retention.go

```diff
@@ -0,0 +1,219 @@
+package retention
+
+import (
+	"context"
+	"errors"
+	"strconv"
+
+	"github.com/go-openapi/runtime"
+	v2client "github.com/mittwald/goharbor-client/apiv2/internal/api/client"
+	"github.com/mittwald/goharbor-client/apiv2/internal/legacyapi/client"
+	"github.com/mittwald/goharbor-client/apiv2/internal/legacyapi/client/products"
+	model "github.com/mittwald/goharbor-client/apiv2/model/legacy"
+	"github.com/mittwald/goharbor-client/apiv2/project"
+	pc "github.com/mittwald/goharbor-client/apiv2/project"
+)
+
+const (
+	AlgorithmOr string = "or"
+
+	// Key for defining matching repositories
+	ScopeSelectorRepoMatches ScopeSelector = "repoMatches"
+
+	// Key for defining excluded repositories
+	ScopeSelectorRepoExcludes ScopeSelector = "repoExcludes"
+
+	// Key for defining matching tag expressions
+	TagSelectorMatches TagSelector = "matches"
+
+	// Key for defining excluded tag expressions
+	TagSelectorExcludes TagSelector = "excludes"
+
+	// The kind of the retention selector, _always_ defaults to 'doublestar'
+	SelectorTypeDefault string = "doublestar"
+
+	// Retain the most recently pushed n artifacts - count
+	PolicyTemplateLatestPushedArtifacts PolicyTemplate = "latestPushedK"
+
+	// Retain the most recently pulled n artifacts - count
+	PolicyTemplateLatestPulledArtifacts PolicyTemplate = "latestPulledN"
+
+	// Retain the artifacts pushed within the last n days
+	PolicyTemplateDaysSinceLastPush PolicyTemplate = "nDaysSinceLastPush"
+
+	// Retain the artifacts pulled within the last n days
+	PolicyTemplateDaysSinceLastPull PolicyTemplate = "nDaysSinceLastPull"
+
+	// Retain always
+	PolicyTemplateRetainAlways PolicyTemplate = "always"
+)
+
+type Client interface {
+	NewRetentionPolicy(ctx context.Context, scopeSelector ScopeSelector, projectRef int64,
```

Comment:
These are a lot of arguments. I don't really know much about the retention stuff, but is it possible to let an library user construct a retention object first and use this as an argument? Given that most of these arguments belong to a retention object of course.